sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_intel_matrix.asciidoc

```diff
@@ -21,135 +19,380 @@
       // ---------------------------------
       // a1, a2, a3, a4, b1, b2, b3, b4, c1, c2, c3, c4, d1, d2, d3, d4
 
-## Supported Combinations Per Hardware
-
-The table below provides a list of the combinations that `joint_matrix` implementations support on each of Intel AMX and Intel XMX hardware. Note that these can be returned in a parametrized way using the `tpu_params` query class.
+=== Example using int8_t type
+```c++
+using namespace sycl::ext::oneapi::experimental::matrix;
+
+queue q;
+range<2> G = {M/tM, N};
+range<2> L = {1, SG_SIZE};
+auto bufA = sycl::buffer{memA, sycl::range{M*K}};
+auto bufB = sycl::buffer{memB, sycl::range{K*N}};
+auto bufC = sycl::buffer{memC, sycl::range{M*N}};
+q.submit([&](sycl::handler& cgh) {
+  auto accA = sycl::accessor{bufA, cgh, sycl::read_only};
+  auto accB = sycl::accessor{bufB, cgh, sycl::read_only};
+  auto accC = sycl::accessor{bufC, cgh, sycl::read_write};
+  cgh.parallel_for(nd_range<2>(G, L), [=](nd_item<2> item)
+  [[sycl::reqd_sub_group_size(SG_SIZE)]] {
+    const auto global_idx = item.get_global_id(0);
+    const auto global_idy = item.get_global_id(1);
+    const auto sg_startx = global_idx - item.get_local_id(0);
+    const auto sg_starty = global_idy - item.get_local_id(1);
+    sub_group sg = item.get_sub_group();
+    joint_matrix<sub_group, int8_t, use::a, tM, tK, layout::row_major> tA;
+    joint_matrix<sub_group, int8_t, use::b, tK, tN,
+                ext::intel::experimental::matrix::layout::packed> tB;
+    joint_matrix<sub_group, int32_t, use::accumulator, tM, tN> tC;
+    joint_matrix_fill(sg, tC, 0);
+    for (int k = 0; k < K; k += tK) {
+      joint_matrix_load(sg, tA, accA + sg_startx * tM * K + k, K);
+      joint_matrix_load(sg, tB, accB + k * N*4 + sg_starty/SG_SIZE*tN*4, N*4);
+      tC = joint_matrix_mad(sg, tA, tB, tC);
+    }
+    auto wi_data_c = ext::intel::experimental::matrix::get_wi_data(sg, tC);
+    for (int i = 0; i < wi_data_c.length(); i++)
+      wi_data_c[i] *= alpha;
+    joint_matrix_store(sg, tC,
+      accC + sg_startx * tM * N + sg_starty/SG_SIZE*tN, N, layout::row_major);
+  });
+});
+q.wait();
+```
 
-### Intel AMX Supported Combinations
+=== Intel-Specific Runtime Query
+Besides the query we provide in
+link:../experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc[sycl_ext_oneapi_matrix],
+some device descriptors are Intel hardware specific. These are
+provided as part of `ext::intel::experimental::info::device::matrix`
+namespace:
 
 [frame="none",options="header"]
 |======================
-| A type | B type | Accumulator type | M | N | K
-| (u)int8_t  | (u)int8_t |  int32_t  |  +<=+ 16 |  +<=+ 16 |  +<=+ 64
-|  bf16       |  bf16   |   fp32   |  +<=+ 16 |  +<=+ 16   |  +<=+ 32
+| Device descriptors | Return type| Description
+|`ext::oneapi::experimental::info::device::matrix::numtiles`| `uint32_t`
+|indicates number of tiles in Intel AMX (does not apply to Intel XMX)
```

Comment:
Yes, zero is the value being used for other things. So I will specify that.
yes, this can be specified in a general manner. A better way would be to talk about the to matrix hardware with separate register files (AMX), in this case: `num_tiles` is used. Otherwise, if the matrix hardware shares same storage with the rest of EUs, then this can be already queried using existing mechanisms.